state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -973,3 +987,67 @@ impl<T: Config> OnKilledAccount<T::AccountId> for Pallet<T> {
 		BoundRedeemAddress::<T>::remove(account_id);
 	}
 }
+
+impl<T: Config> SubAccountHandler<T::AccountId> for Pallet<T> {
+	fn derive_sub_account(
+		parent_account_id: T::AccountId,
+		sub_account_index: u8,
+	) -> Result<T::AccountId, DispatchError> {
+		let sub_account_id: T::AccountId = Decode::decode(&mut TrailingZeroInput::new(
+			(*b"chainflip/subaccount", parent_account_id.clone(), sub_account_index)
+				.blake2_256()
+				.as_ref(),
+		))
+		.map_err(|_| Error::<T>::SubAccountIdDerivationFailed)?;
+
+		Ok(sub_account_id)
+	}
+
+	fn derive_and_fund_sub_account(
+		parent_account_id: T::AccountId,
+		sub_account_index: u8,
+	) -> Result<T::AccountId, DispatchError> {
```

Comment:
I think it would be useful to allow specifying the starting balance of the subaccount instead of assuming `MinimumFunding`.